packages/web-components/src/pane/pane.ts

```diff
@@ -0,0 +1,260 @@
+import { attr, FASTElement, observable, Updates } from '@microsoft/fast-element';
+import { isHTMLElement, keyEscape, keyTab } from '@microsoft/fast-web-utilities';
+import { PanePosition, PaneSize } from './pane.options.js';
+
+export interface OpenEvent {
+  open: boolean;
+  position?: string;
+  controlSize?: PaneSize | number;
+}
+
+// The Pane class extends the FASTElement which provides a foundation for creating
+// custom elements using FAST. It extends the HTMLElement interface of the browser.
+export class Pane extends FASTElement {
+  private panes: Pane[] = [];
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    // Register event listener
+    this.addEventListener('keydown', this.handleKeyDown);
+    this.setItems();
+  }
+
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+    // Unregister event listener
+    this.removeEventListener('keydown', this.handleKeyDown);
+  }
+
+  private hasInteracted: boolean = false;
+
+  // The previous active element before opening the pane.
+  private previousActiveElement?: HTMLElement;
+
+  /**
+   * The pane element.
+   * @public
+   * @remarks
+   * HTML Attribute: pane
+   */
+  @observable
+  public _pane?: HTMLElement;
+
+  /**
+   * Determines whether the pane should be displayed as modal or non-modal.
+   * When in modal mode, an overlay is applied over the rest of the view.
+   * @public
+   * @remarks
+   * HTML Attribute: modal
+   */
+  @attr({ mode: 'boolean' })
+  public modal: boolean = false;
+
+  /**
+   * Sets the position of the pane (left/right).
+   * @public
+   * @remarks
+   * HTML Attribute: position
+   * @defaultValue right
+   */
+  @attr
+  public position?: PanePosition;
+
+  /**
+   * The maximum number of options to display.
+   *
+   * @remarks
+   * HTML Attribute: `size`.
+   *
+   * @public
+   */
+  @attr({ attribute: 'control-size' })
+  public controlSize?: PaneSize | number | undefined;
+
+  /**
+   * Indicates whether the pane is open or closed.
+   * @public
+   * @remarks
+   * HTML Attribute: open
+   */
+  @observable
+  @attr({ mode: 'boolean' })
+  public open: boolean = false;
+
+  /**
+   * Indicates the presence of the toolbar.
+   * @public
+   * @remarks
+   * HTML Attribute: toolbar
+   */
+  @attr({ mode: 'boolean' })
+  public toolbar?: boolean = false;
+
+  /**
+   * The element to receive focus when the pane opens.
+   * @public
+   * @remarks
+   * HTML Attribute: focus-target
+   */
+  @attr({ attribute: 'focus-target' })
+  public focusTarget?: string;
```

Comment:
Why does the component need to handle this if the children are provided by someone else? If this isn't internal to this component this feels like unnecessary overhead.